src/main/java/com/google/api/gax/grpc/PageStreamingCallable.java

```diff
@@ -118,5 +132,40 @@ public void remove() {
         throw new UnsupportedOperationException();
       }
     }
+
+    @Override
+    public Iterable<ResourceT> getPageValues() {
+      return pageDescriptor.extractResources(getPage());
+    }
+
+    @Override
+    public PageAccessor<ResourceT> getNextPage() {
+      Object nextToken = getNextPageToken();
+      if (nextToken == null) {
+        return null;
+      } else {
+        RequestT nextRequest =
+            pageDescriptor.injectToken(context.getRequest(), getNextPageToken());
+        return new PageAccessorImpl<>(callable, pageDescriptor,
+                                      context.withRequest(nextRequest));
+      }
+    }
+
+    @Override
+    public String getNextPageToken() {
+      Object nextToken = pageDescriptor.extractNextToken(getPage());
+      if (nextToken.equals(pageDescriptor.emptyToken())) {
+        return null;
+      } else {
+        return nextToken.toString();
+      }
+    }
+
+    private ResponseT getPage() {
+      if (currentPage == null) {
+        currentPage = Futures.getUnchecked(callable.futureCall(context));
```

Comment:
Right now we expose ApiException but I am not sure if we should expose everything else.
Based on the documentation, UncheckedExecutionException coverts all the underlying exceptions so the upper-level exception handler could be simpler. I am slightly inclined not to break this principle since our ApiCallable is suppose to be consistent with Futures.